praw/objector.py

```diff
@@ -24,6 +46,56 @@ def kind(self, instance):
             if isinstance(instance, self.parsers[key]):
                 return key
 
+    def _objectify_dict(self, data):
+        """Create RedditBase objects from dicts.
+
+        :param data: The structured data, assumed to be a dict.
+        :returns: An instance of :class:`~.RedditBase`.
+
+        """
+        if ({'conversation', 'messages', 'user', 'modActions'}.issubset(data)):
+            parser = self.parsers['ModmailConversation']
+            return parser.parse(data, self._reddit)
```

Comment:
With the code written this way, we should be able to have just one `return parser.parse(data, self._reddit)` at the end outside of the conditional, since the only different case is when we're directly returning the dict (the else case).

You may need to initialize `parser = None` before the conditionals so the static analysis tool doesn't complain. After that we should be good to go.